### DISCUSSION TITLE
Is there any way to send push notification using supabase?

### DISCUSSION THREAD

**Discussion by wrteam-priyansh** [supabase/supabase#3588] [Questions] [answered]
Is there any way to send push notifications to a given user or set of users after some changes happen in the database? Like user places an order so the seller should get notification of new order.

**Comment by dshukertjr** [comment, marked as answer]
Hi @wrteam-priyansh!

You can set up [Supabase Function Hooks](https://supabase.io/blog/2021/07/30/supabase-functions-updates) to receive webhook requests when there is a change in your database. Using this, you can notify external servers like Firebase Cloud Functions endpoint to trigger push notifications using FCM.

**Reply by DevMensIT** [reply]
hey i am not able to find webhook on supabase.

**Reply by dshukertjr** [reply]
@TheDevManIT I found [this](https://supabase.com/blog/2021/07/30/supabase-functions-updates) article helpful, but were you not able to follow this within your project? If you could provide more detail on where you are having trouble, I might be able to help you out!

**Reply by DevMensIT** [reply]
There is no "function hook " on my project.

**Reply by GaryAustin1** [reply]
Is this on an older project before July of last year or so.  Function Hooks were only for projects created after that.
Are you selfhosted?  It is possible Studio does not support that yet, like storage.

**Reply by DevMensIT** [reply]
May be that's older project. What should i do now ? Is "Function Hook" available on free tier ?

**Reply by GaryAustin1** [reply]
It is on free tier.

**Reply by DevMensIT** [reply]
Should i create new project and check ?

**Reply by dshukertjr** [reply]
@TheDevManIT Creating an new project might be an option if you don't mind that. Otherwise, you can contact Supabase support about not seeing Function Hook within your console. 
<img width="1380" alt="Screen Shot 2022-03-29 at 22 10 04" src="https://user-images.githubusercontent.com/18113850/160618732-5ac86795-9d0c-4452-b3be-54e135ba72c2.png">

**Reply by DevMensIT** [reply]
I created new projects and guess what i can see function hooks there 
how can i get in my old project ?

**Reply by martin-braun** [reply]
I'm interested in Supabase, but this is a bummer for me. The whole reason why I got interested in Supabase is the ability to free my customers from Google when it comes to their data. So recommend to use FCM is no option for me.

**Reply by GaryAustin1** [reply]
@martin-braun This is a bit old discussion.  Supabase has edge-functions now that did not exist when this was posted.
https://supabase.com/docs/guides/functions

**Reply by martin-braun** [reply]
@GaryAustin1 Thanks, +1 for using Deno, very cool. I am interested in using [Directus with Supabase](https://supabase.com/docs/guides/integrations/directus), so missing webhooks are not the problem. I was hoping Supabase offers a seamless experience alternative to FCM, but I learned that this is only possible by using an own push client/server which defeats the purpose. I miss to have at least a proper push server that uses FCM (Android), APN (iOS) or maybe even alternative push clients (by using [UnifiedPush](https://unifiedpush.org)) embedded in Supabase that I can reach from my webhooks in Directus.

**Reply by Wamy-Dev** [reply]
> I'm interested in Supabase, but this is a bummer for me. The whole reason why I got interested in Supabase is the ability to free my customers from Google when it comes to their data. So recommend to use FCM is no option for me. 

I dont really think Supabase can ever have an alternative due to Android relying on FCM in the background to send all notifications on the OS. Any solution that Supabase can come up with will require FCM anyways. Same with OneSignal, who also has to use FCM to send notifications to Android users.

**Comment by wrteam-priyansh** [comment]
@dshukertjr Is it possible to have simple working example for supabse function?

**Reply by dshukertjr** [reply]
@wrteam-priyansh You can follow [this](https://www.youtube.com/watch?v=codAs9-NeHM) video for an actual live example calling stripe API using Supabase function hooks!

**Reply by Tristan-2021** [reply]
Hi Tyler, is this possible on the flutter side? send notifications in foreground and background?

**Comment by ditcu-code** [comment]
has anyone managed to do this with edge functions?

**Reply by dshukertjr** [reply]
You can follow this guide to see how you can implement it. 
https://supabase.com/docs/guides/integrations/onesignal

**Comment by Yogesh-Dubey-Ayesavi** [comment]
I use rpc functions with triggers to manage this.
